In [2]:
# 0. 모듈 불러오기

import pandas as pd
import numpy as np

In [3]:
# 1. 데이터 불러오기

train_data = pd.read_json('data/train.json', encoding='UTF-8', typ = 'frame')
song_meta = pd.read_json('data/song_meta.json', encoding='UTF-8', typ = 'frame')
genre_gn_all = pd.read_json('data/genre_gn_all.json',encoding='UTF-8', typ = 'series')
genre_gn_all = pd.DataFrame(genre_gn_all, columns = ['gnr_name']).reset_index().rename(columns = {'index' : 'gnr_code'})

### 데이터를 확인해보자!

In [4]:
train_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 115071 entries, 0 to 115070
Data columns (total 6 columns):
 #   Column        Non-Null Count   Dtype 
---  ------        --------------   ----- 
 0   tags          115071 non-null  object
 1   id            115071 non-null  int64 
 2   plylst_title  115071 non-null  object
 3   songs         115071 non-null  object
 4   like_cnt      115071 non-null  int64 
 5   updt_date     115071 non-null  object
dtypes: int64(2), object(4)
memory usage: 5.3+ MB


In [5]:
song_meta.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 707989 entries, 0 to 707988
Data columns (total 9 columns):
 #   Column                  Non-Null Count   Dtype 
---  ------                  --------------   ----- 
 0   song_gn_dtl_gnr_basket  707989 non-null  object
 1   issue_date              707989 non-null  int64 
 2   album_name              707985 non-null  object
 3   album_id                707989 non-null  int64 
 4   artist_id_basket        707989 non-null  object
 5   song_name               707989 non-null  object
 6   song_gn_gnr_basket      707989 non-null  object
 7   artist_name_basket      707989 non-null  object
 8   id                      707989 non-null  int64 
dtypes: int64(3), object(6)
memory usage: 48.6+ MB


In [6]:
genre_gn_all.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 254 entries, 0 to 253
Data columns (total 2 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   gnr_code  254 non-null    object
 1   gnr_name  254 non-null    object
dtypes: object(2)
memory usage: 4.1+ KB


#### train_data 크기 : 115071 (약 11만)
#### song 크기 : 707989 (약 70만)
#### genre_gn_all : 254

In [6]:
train_data.head()

tags      id  \
0                                                [락]   61281   
1                                           [추억, 회상]   10532   
2                                          [까페, 잔잔한]   76951   
3  [연말, 눈오는날, 캐럴, 분위기, 따듯한, 크리스마스캐럴, 겨울노래, 크리스마스,...  147456   
4                                               [댄스]   27616   

            plylst_title                                              songs  \
0                여행같은 음악  [525514, 129701, 383374, 562083, 297861, 13954...   
1                요즘 너 말야  [432406, 675945, 497066, 120377, 389529, 24427...   
2  편하게, 잔잔하게 들을 수 있는 곡.-  [83116, 276692, 166267, 186301, 354465, 256598...   
3  크리스마스 분위기에 흠뻑 취하고 싶을때  [394031, 195524, 540149, 287984, 440773, 10033...   
4               추억의 노래 ㅋ  [159327, 553610, 5130, 645103, 294435, 100657,...   

   like_cnt                updt_date  
0        71  2013-12-19 18:36:19.000  
1         1  2014-12-02 16:19:42.000  
2        17  2017-08-28 07:09:34.000  
3        33  2019-12-05 15:15:18.000  
4         9  2011-10-25 13:54:56.000

In [11]:
song_meta['id'].describe()

count    707989.000000
mean     353994.000000
std      204378.964203
min           0.000000
25%      176997.000000
50%      353994.000000
75%      530991.000000
max      707988.000000
Name: id, dtype: float64

In [12]:
train_data['id'].describe()

count    115071.000000
mean      76728.259501
std       44313.134619
min           1.000000
25%       38277.500000
50%       76732.000000
75%      115061.500000
max      153428.000000
Name: id, dtype: float64

In [7]:
genre_gn_all

gnr_code gnr_name
0     GN0100      발라드
1     GN0101   세부장르전체
2     GN0102      '80
3     GN0103      '90
4     GN0104      '00
..       ...      ...
249   GN2900      뮤지컬
250   GN2901   세부장르전체
251   GN2902    국내뮤지컬
252   GN2903    국외뮤지컬
253   GN3000    크리스마스

[254 rows x 2 columns]

## processed 된 Data load
#### carol들로 이루어진 플레이리스트 + carol 들

In [21]:
import import_ipynb
from util import load_json

train_playlist = load_json("./data/processedData/playlist_train.json")
val_playlist = load_json("./data/processedData/playlist_val.json")
test_playlist = load_json("./data/processedData/playlist_test.json")
carol_data = load_json("./data/processedData/carol_data.json")
train_playlist = pd.DataFrame(eval(train_playlist))
val_playlist = pd.DataFrame(eval(val_playlist))
test_playlist = pd.DataFrame(eval(test_playlist))
carol_data = pd.DataFrame(eval(carol_data))

In [27]:
print(len(train_playlist))
train_playlist.head()

1030


tags      id  \
0  [연말, 분위기, 겨울, 성탄절, Christmas, 기분전환, 크리스마스, 캐롤,...   67590   
1                                           [솔로, 눈물]   67478   
2         [겨울, 겨울감성, 캐럴, 아이돌, 걸그룹, 크리스마스, 보이그룹, 신나는]  135680   
3                                  [명곡, 홀리데이, 크리스마스]   71752   
4  [기분좋은, 연말, 감성, 크리스마스이브, 달달한, 성탄절, 크리스마스노래, 설렘노...   40876   

               plylst_title  \
0        크리스마스 분위기 같이 느껴볼래?   
1               솔로 크리스마스..    
2  포근한 크리스마스 분위기 더해줄 아이돌 캐럴   
3                   크리스마스니까   
4         나만의 크리스마스 느낌 노래들_   

                                               songs  like_cnt  \
0  [404160, 424080, 172734, 677560, 387859, 22667...        16   
1  [546793, 547185, 272539, 253755, 425758, 32911...         8   
2  [262202, 112399, 187149, 326204, 345728, 16197...         6   
3  [704789, 568514, 18973, 79517, 667770, 139082,...         8   
4  [387859, 226670, 359615, 416029, 653711, 53504...         7   

                 updt_date  
0  2019-12-25 10:32:37.000  
1  2016-12-24 02:50:30.000  
2  2019-11-26 11:12:40.000  
3  2018-12-20 15:10:57.000  
4  2018-03-21 22:08:52.000

In [28]:
carol_data.head()

song_gn_dtl_gnr_basket  issue_date                          album_name  \
0  [GN1601, GN1615, GN1610]    20171006      Christmas at Westminster Abbey   
1  [GN1701, GN1703, GN1702]    20121109  Christmas (Deluxe Special Edition)   
2                  [GN0101]    20190408                                있어줄래   
3          [GN0105, GN0101]    20121204                       WINTER POETRY   
4          [GN1304, GN1301]    20110810                        Simple Steps   

   album_id         artist_id_basket  \
0  10104201  [345328, 36652, 347399]   
1   2167838                  [35370]   
2  10271322                 [500421]   
3   2169041                   [1741]   
4   2058451                 [614091]   

                                           song_name song_gn_gnr_basket  \
0  Anonymous: Up! Awake! From Highest Steeple - A...           [GN1600]   
1          Mis Deseos \/ Feliz Navidad (with Thalia)           [GN1700]   
2                                               있어줄래           [GN0100]   
3                                       너에게 전하지 못한 말           [GN0100]   
4                                      Rollercoaster           [GN1300]   

                                  artist_name_basket  id  
0  [Choir Of Westminster Abbey, Simon Preston, Ch...  23  
1                                    [Michael Buble]  41  
2                                             [길구봉구]  54  
3                                              [신혜성]  70  
4                                          [Sam Ock]  71

In [29]:
print(len(val_playlist))
val_playlist.head()

258


tags      id  \
0                                         [겨울, 기분전환]   70387   
1  [여유로운, 겨울, 휴식, 매장음악, 카페, 잔잔한, 힐링, 기분전환, 크리스마스,...   59811   
2                                       [겨울, 설렘, 사랑]   17664   
3  [매장음악, 캐럴, 카페, 상쾌, LINYL, 기분전환, 크리스마스, 캐롤, 즐거운...   17426   
4                                 [눈, 겨울, 크리스마스, 캐롤]  121806   

                  plylst_title  \
0  크리스마스분위기에 흠뻑~겨울냄새 물씬!크리스마스송   
1       선물과 같은 크리스마스에 듣는 연주곡:)   
2            크리스마스가 빨리 왔으면 좋겠다   
3    매 년 새로워, 짜릿해! 크리스마스 (POP)   
4       내 맘에 쏙드는 크리스마스 노래모음..✿   

                                               songs  like_cnt  \
0  [387859, 112399, 10445, 326204, 345728, 249436...         3   
1  [261726, 6984, 663277, 366237, 205608, 545671,...         0   
2  [438403, 185874, 74551, 318487, 551335, 623455...        17   
3  [387859, 271057, 422873, 526035, 433118, 35635...         8   
4  [443077, 136428, 595353, 187149, 326204, 56799...         3   

                 updt_date  
0  2016-12-02 19:01:43.000  
1  2019-12-23 07:31:13.000  
2  2015-10-11 09:41:02.000  
3  2019-12-06 16:46:14.000  
4  2017-11-24 15:47:01.000

In [30]:
print(len(test_playlist))
test_playlist.head()

322


tags      id  \
0                                         [휴식, 잔잔한]   72753   
1                     [겨울, 겨울노래베스트, 로맨틱, 크리스마스, 캐롤]  132953   
2       [겨울, 로맨틱, 메리크리스마스, 연인, 분위기, 겨울분위기, 로맨스, 캐롤]   15133   
3  [연말, 연주곡, 피아노, 꿀잠, 잠들기전, 캐롤, 부드러운, 자장가, 숙면, 따뜻한]   65372   
4        [겨울, 분위기, 카페, 케롤, 첫눈, 크리스마스, 설렘, 거리, 겨울감성]   85669   

                    plylst_title  \
0                       크리스마스239   
1  미리 크리스마스~❤ 달달하고 로맨틱한 겨울감성 뮤직!   
2             메리 크리스마스 흔하지 않은 캐롤   
3        잠이 솔솔 올때 듣는 음악(캐롤,뉴에이지)   
4       크리스마스날 길에서 나오면 설레는 노래>_<   

                                               songs  like_cnt  \
0  [405485, 381653, 566122, 546313, 158539, 68697...         0   
1  [196522, 305218, 61771, 338900, 303055, 248418...         9   
2  [137754, 166884, 321147, 312613, 327471, 33289...         9   
3  [121837, 333474, 276541, 422708, 301555, 22007...         4   
4  [387859, 265904, 112399, 326204, 695032, 62277...        16   

                 updt_date  
0  2019-11-19 13:35:09.000  
1  2018-11-27 23:08:20.000  
2  2017-11-09 17:47:27.000  
3  2018-12-12 17:42:27.000  
4  2020-02-28 14:06:37.000